server/matchmaker/map_pool.py

```diff
@@ -24,39 +26,96 @@ def __init__(
+                transfer_weight_proportionally(map_id, weight_transfer_targets)
+
+        return adjusted_weights
+
+    def choose_map(self, played_map_ids: Iterable[int] = (), initial_weights: Optional[dict[int, float]] = None) -> Map:
+        """
+            Selects a random map using veto system weights with an anti-repetition adjustment.
         """
         if not self.maps:
-            self._logger.critical(
-                "Trying to choose a map from an empty map pool: %s", self.name
-            )
+            self._logger.critical("Trying to choose a map from an empty map pool: %s", self.name)
             raise RuntimeError(f"Map pool {self.name} not set!")
 
-        # Make sure the counter has every available map
-        counter = Counter(self.maps.keys())
-        counter.update(id_ for id_ in played_map_ids if id_ in self.maps)
+        self._logger.debug("______initial_played_map_ids___________: %s", list(played_map_ids))
+        played_map_pool_map_version_ids = [
+            self.maps[id].map_pool_map_version_id
+            for id in played_map_ids
+            if id in self.maps
+        ]
+        self._logger.debug("______played_map_pool_map_version_ids_________: %s", played_map_pool_map_version_ids)
 
-        least_common = counter.most_common()[::-1]
-        least_count = 1
-        for id_, count in least_common:
-            if isinstance(self.maps[id_], Map):
-                least_count = count
-                break
+        map_list = [(m.map_pool_map_version_id, m) for m in self.maps.values()]
 
-        # Trim off the maps with higher play counts
-        for i, (_, count) in enumerate(least_common):
-            if count > least_count:
-                least_common = least_common[:i]
-                break
+        if initial_weights is None:
+            initial_weights = {mp_mv_id: 1.0 for mp_mv_id, _ in map_list}
 
-        weights = [self.maps[id_].weight for id_, _ in least_common]
-
-        map_id = random.choices(least_common, weights=weights, k=1)[0][0]
-        return self.maps[map_id].get_map()
+        adjusted_weights = self.apply_antirepetition_adjustment(
+            initial_weights, played_map_pool_map_version_ids, config.LADDER_ANTI_REPETITION_WEIGHT_BASE_THRESHOLDS, config.LADDER_ANTI_REPETITION_REPEAT_COUNTS_FACTOR
+        )
+        self._logger.debug("______adjusted_weights________________: %s", adjusted_weights)
+        self._logger.debug("______map_list________________: %s", map_list)
+        final_weights = [adjusted_weights.get(mp_mv_id, 0) * m.weight for mp_mv_id, m in map_list]
+        self._logger.debug("______final_weights________________: %s", final_weights)
+        return random.choices([map for _, map in map_list], weights=final_weights, k=1)[0].get_map()  # nosec B311
 
     def __repr__(self) -> str:
         return f"MapPool({self.id}, {self.name}, {list(self.maps.values())})"
+
+
+class MatchmakerQueueMapPool(NamedTuple):
```

Comment:
It's not obvious to me what this class is supposed to do. I think to understand that I'd have to read the full PR first.

It seems to connect a map pool with information that the veto system needs. Why do we need this class? Why is this not either a part of the MapPool or of the MatchmakerQueue itself? 

If we think we need this class, then please improve:
- The naming of the class. The current name concatenates the names of two existing classes. But how does this relate to MatchmakerQueue-s? That's not obvious to me.
- give it a docstring